### DISCUSSION TITLE
iPhone Tethering Instead of LTE Modem

### DISCUSSION THREAD

**Discussion by focher** [tesla-android/issue-tracker#67] [General]
I managed to get iPhone tethering on the Linux Pi working to provide Internet connectivity instead of the LTE modem. I installed usbmuxd (`sudo apt-get install usbmuxd`), rebooted, then plugged the iPhone in with a USB A to Lightning cable with the Personal Hotspot enabled in Settings.

This allows the Tesla to connect to the Tesla-Android wifi network and get an IP because it passes the check that there is Internet connectivity.

**Comment by mikegapinski** [comment]
The internet connectivity is not required starting with version 2022.27.1. Tethering from your iPhone is coming as a feature in the single board stack

**Reply by focher** [reply]
That's great! I've upgraded and the connection now works fine. Other issue, but I'll post a different discussion.

**Comment by Cooker44** [comment]
Thanks for this, have installed same and tethered.  No need for LTE modem to maintain car connectivity to internet! That’s saved me £40 and £8pm…thank you!

**Reply by Cooker44** [reply]
Could really do with this in single board release please @mikegapinski

**Comment by RetoHB** [comment]
Does this still work with the single board to use Internet data from the phone?
Or will this be included in the October 31st release? Possibly via Wifi or a second Wifi dongle on the Pi4?

**Comment by tomvanzummeren** [comment]
I would love to have this too. Is it already known which release will enable this? Or can I do it myself by installing usbmuxd? The problem is I have no idea how to access the terminal while Tesla Android is installed (I used the single image method with a 64gb sd card)

**Reply by mikegapinski** [reply]
USB tethering from iOS is not yet there on the single board stack.

**Reply by tomvanzummeren** [reply]
Ah so installing usbmuxd requires the two board stack? Gotcha. Any idea when the version will be released of the single board stack that supports iPhone tethering over USB?

**Reply by mikegapinski** [reply]
Soon, Android USB tethering goes live in the next build. It also has the kernel driver for iOS tethering, I just need to get usbmuxd integrated with Android, that's not a common, documented use case for it

**Reply by tomvanzummeren** [reply]
Awesome! Well, I wish you the best of luck figuring it out 😊

**Comment by RetoHB** [comment]
USB tethering works fine in 2022.45.1. However it seems the car itself cannot access internet through Tesla Android and USB tethering. Is this a setting I can change in Tesla Android?
